integration/entity.go

```diff
@@ -30,6 +30,26 @@ type EntityMetadata struct {
 	IDAttrs   IDAttributes `json:"id_attributes"` // For entity Key uniqueness
 }
 
+// EqualsTo returns true when both metadata are equal.
+func (m *EntityMetadata) EqualsTo(b *EntityMetadata) bool {
+	// prevent checking on Key() for performance
```

Comment:
I just wonder if really performance matters in this comparison. If I'm not wrong `integration.Entity` method is usually called at the beginning of the integration and it does not affect in performance.

In this case we could avoid all this `EqualsTo` & `sameAs` code and just use entity Key comparison here:
https://github.com/newrelic/infra-integrations-sdk/pull/179/files#diff-34fc2068cceaa93f8d6b312c44397541R159
```
e.Key() == eIt.Key()
```

Reply:
I think it matters as we don't know how integrations behave.

For instance an integration could be using `integration.Entity()` method to fetch entities. Instead of creating and keeping them in memory.

Also integration runtime is brought up and teared down on each integration run, so this method is going to be called frequently.

Reply:
fair enough